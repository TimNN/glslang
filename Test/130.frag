#version 130

lowp vec3 a;
mediump float b;
highp int c;

precision highp float;

in vec4 i;
out vec4 o;

flat in float fflat;
smooth in float fsmooth;
noperspective in float fnop;

void main()
{
    float clip = gl_ClipDistance[3];
}

uniform samplerCube sampC;

void foo()
{
    vec4 s = textureGather(sampC, vec3(0.2));
}

#extension GL_ARB_texture_gather : enable

void bar()
{
    vec4 s = textureGather(sampC, vec3(0.2));
}

flat in vec3 gl_Color;     // ERROR, type
in vec4 gl_Color;
flat in vec4 gl_Color;
flat in vec4 gl_Color[2];  // ERROR, array 
vec4 gl_Color;             // ERROR, storage

#extension GL_ARB_texture_gather : warn

void bar2()
{
    vec4 s = textureGather(sampC, vec3(0.2));
}
