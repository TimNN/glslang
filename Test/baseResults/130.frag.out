Warning, version 130 is not yet complete; most features are present, but a few are missing.
ERROR: 0:25: 'texture gather function' : not supported for this version or the enabled extensions 
ERROR: 0:35: 'redeclaration' : cannot change the type of gl_Color
ERROR: 0:38: 'gl_Color' : redeclaring non-array as array 
ERROR: 0:39: 'redeclaration' : cannot change storage, memory, or auxiliary qualification of gl_Color
WARNING: 0:45: extension GL_ARB_texture_gather is being used for texture gather function
ERROR: 4 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:16  Function Definition: main( (void)
0:16    Function Parameters: 
0:18    Sequence
0:18      Sequence
0:18        move second child to first child (float)
0:18          'clip' (float)
0:18          direct index (smooth float)
0:18            'gl_ClipDistance' (smooth in unsized array of float)
0:18            Constant:
0:18              3 (const int)
0:23  Function Definition: foo( (void)
0:23    Function Parameters: 
0:25    Sequence
0:25      Sequence
0:25        move second child to first child (4-component vector of float)
0:25          's' (4-component vector of float)
0:25          Function Call: textureGather(sC1;vf3; (4-component vector of float)
0:25            'sampC' (uniform samplerCube)
0:25            Constant:
0:25              0.200000
0:25              0.200000
0:25              0.200000
0:30  Function Definition: bar( (void)
0:30    Function Parameters: 
0:32    Sequence
0:32      Sequence
0:32        move second child to first child (4-component vector of float)
0:32          's' (4-component vector of float)
0:32          Function Call: textureGather(sC1;vf3; (4-component vector of float)
0:32            'sampC' (uniform samplerCube)
0:32            Constant:
0:32              0.200000
0:32              0.200000
0:32              0.200000
0:43  Function Definition: bar2( (void)
0:43    Function Parameters: 
0:45    Sequence
0:45      Sequence
0:45        move second child to first child (4-component vector of float)
0:45          's' (4-component vector of float)
0:45          Function Call: textureGather(sC1;vf3; (4-component vector of float)
0:45            'sampC' (uniform samplerCube)
0:45            Constant:
0:45              0.200000
0:45              0.200000
0:45              0.200000
0:?   Linker Objects
0:?     'a' (3-component vector of float)
0:?     'b' (float)
0:?     'c' (int)
0:?     'i' (smooth in 4-component vector of float)
0:?     'o' (out 4-component vector of float)
0:?     'fflat' (flat in float)
0:?     'fsmooth' (smooth in float)
0:?     'fnop' (noperspective in float)
0:?     'gl_ClipDistance' (smooth in unsized array of float)
0:?     'sampC' (uniform samplerCube)
0:?     'gl_Color' (smooth in 4-component vector of float)
0:?     'gl_Color' (flat in 4-component vector of float)

